Library/Homebrew/formula.rb

```diff
@@ -405,11 +413,19 @@ def linked_keg
     Pathname.new("#{HOMEBREW_LIBRARY}/LinkedKegs/#{name}")
   end
 
+  def latest_head_prefix
+    head_prefixes = installed_prefixes.select do |pn|
+      pn.basename.to_s.start_with?("HEAD")
```

Comment:
Or, we may consider to split it into two methods: `latest_head_version` and `latest_head_prefix`. Because we still need to check revision number inside `head_version_up_to_date?`